### PR TITLE
Updates on Statistics and Probability, Linear Regression 1, Logistic Regression, K Means Clustering 1

### PR DESCRIPTION
**Pull request by chooneung**
**Statistics and Probability** - formatted some documentations.
**Linear Regression 1** - add explanations and example.
**Logistic Regression** - add explanations.
**K Means Clustering 1** - add explanations and example.

### CODE REVIEW

**Comment by kenghooi-teoh** [comment]
**Logistic Regression**
- explain macro avg, weighted avg, support in confusion matrix
- brief explanation of features you use for modeling (how many, what are they? etc), class distribution - check guan yu's branch he printed very clear description of the dataset
- add multiclass classification example
- explain model coefficients, logits

**K-Means**
- is there any patterns (common charactersitics) in the clusters formed?

**Linear Regression**
- display all feature info
- interpret the model coefficients (how a particular feature affects the output)
- interpret MAE, MSE, RMSE (which one to use? or how to choose between these 3 metrics)

**Prob/Stats**
- provide a few examples of how these concepts are used in ML/DL e.g. weight initialization

**Comment by chooneung** [comment]
@kenghooi-teoh updated, please review thanks =)
